projects/ADuCM360_demo_cn0411/src/CN0411.c

```diff
@@ -0,0 +1,1876 @@
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_compute_tds(cn0411_device *cn0411_dev)
+{
+
+	int32_t ret;
+
+	ret = CN0411_compensate_cond(cn0411_dev);
+	cn0411_dev->tds = cn0411_dev->solution.tds_factor * cn0411_dev->comp_cond;
+
+	return ret;
+}
+
+/**
+ * CN0411 ADC channels internal software calibration
+ *
+ * @param cn0411_dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_ADC_int_calibrate(cn0411_device *cn0411_dev)
+{
+	int32_t ret;
+
+	ret = CN0411_ADC_ch0 (cn0411_dev, ADC_CH_ENABLE);
+	ret |= CN0411_ADC_operation_mode(cn0411_dev, CAL_INT_ZERO_MODE);
+	timer_sleep(1000u);
+	ret |= CN0411_ADC_ch0 (cn0411_dev, ADC_CH_DISABLE);
+	ret |= CN0411_ADC_ch1 (cn0411_dev, ADC_CH_ENABLE);;
+	ret |= CN0411_ADC_operation_mode(cn0411_dev, CAL_INT_ZERO_MODE);
+	timer_sleep(1000u);
+	ret |= CN0411_ADC_ch1 (cn0411_dev, ADC_CH_DISABLE);
+	ret |= CN0411_ADC_ch2 (cn0411_dev, ADC_CH_ENABLE);
+	ret |= CN0411_ADC_operation_mode(cn0411_dev, CAL_INT_ZERO_MODE);
+	timer_sleep(1000u);
+	ret |= CN0411_ADC_ch2 (cn0411_dev, ADC_CH_DISABLE);
+
+	return ret;
+}
+
+/**
+ * CN0411 ADC channels system calibration
+ *
+ * @param cn0411_dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_ADC_sys_calibrate(cn0411_device *cn0411_dev)
+{
+	int32_t ret;
+
+	ret = CN0411_ADC_operation_mode(cn0411_dev, IDLE_MODE);
+	ret |= CN0411_ADC_ch0 (cn0411_dev, ADC_CH_DISABLE);
+	ret |= CN0411_ADC_ch1 (cn0411_dev, ADC_CH_DISABLE);
+	ret |= CN0411_ADC_ch2 (cn0411_dev, ADC_CH_DISABLE);
+	pwm_status = PWM_SYSCALIB_AIN7;
+	ret |= CN0411_ADC_ch1 (cn0411_dev, ADC_CH_ENABLE);
+	timer_sleep(1000u);
+	ret |= CN0411_ADC_operation_mode(cn0411_dev, CAL_SYS_ZERO_MODE);
+	timer_sleep(1000u);
+	ret |= CN0411_ADC_ch1 (cn0411_dev, ADC_CH_DISABLE);
+	pwm_status = PWM_SYSCALIB_AIN8;
+	ret |= CN0411_ADC_ch2 (cn0411_dev, ADC_CH_ENABLE);
+	timer_sleep(1000u);
+	ret |= CN0411_ADC_operation_mode(cn0411_dev, CAL_SYS_ZERO_MODE);
+	timer_sleep(1000u);
+	ret |= CN0411_ADC_ch2 (cn0411_dev, ADC_CH_DISABLE);
+	pwm_status = PWM_CONVERSION;
+	CN0411_ADC_conv_init (cn0411_dev, cn0411_dev->conv_type);
+
+	return ret;
+}
+
+/**
+ * CN0411 Premeasurement Process
+ *
+ * Autoset gain resistor channel based on the read DAC value
+ *
+ * @param cn0411_dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_premeasurement(cn0411_device *cn0411_dev)
+{
+	int32_t ret;
+
+	cn0411_dev->ch_gain = CH_GAIN_RES_20M;
+	ret = CN0411_read_vdac(cn0411_dev);
+	cn0411_dev->v_exc = cn0411_dev->read_dac;
+	ret |= CN0411_DAC_set_value(cn0411_dev, cn0411_dev->v_exc);
+	while (cn0411_dev->ch_gain >= 1) {
+		ret |= CN0411_ADC_set_io1 (cn0411_dev, cn0411_dev->ch_gain);
+		ret |= CN0411_read_vpp(cn0411_dev);
+		if((cn0411_dev->vpp > 0.3 * 2 * cn0411_dev->v_exc)
+		    || (cn0411_dev->ch_gain == 1)) {
```

Comment:
try to fit this in one line:
`if((cn0411_dev->vpp > 0.3 * 2 * cn0411_dev->v_exc) || (cn0411_dev->ch_gain == 1)) {`